pilot/cmd/pilot-agent/status/server.go

```diff
@@ -584,22 +651,21 @@ func getHeaderTimeout(timeout string) (time.Duration, error) {
 // scrape will send a request to the provided url to scrape metrics from
 // This will attempt to mimic some of Prometheus functionality by passing some of the headers through
 // such as accept, timeout, and user agent
-// Returns the scraped metrics as well as the response's "Content-Type" header to determine the metrics format
-func (s *Server) scrape(url string, header http.Header) ([]byte, string, error) {
+// Returns the scraped metrics reader as well as the response's "Content-Type" header to determine the metrics format
+func (s *Server) scrape(url string, header http.Header) (io.ReadCloser, context.CancelFunc, string, error) {
+	var cancel context.CancelFunc
 	ctx := context.Background()
 	if timeoutString := header.Get("X-Prometheus-Scrape-Timeout-Seconds"); timeoutString != "" {
 		timeout, err := getHeaderTimeout(timeoutString)
 		if err != nil {
 			log.Warnf("Failed to parse timeout header %v: %v", timeoutString, err)
 		} else {
-			c, cancel := context.WithTimeout(ctx, timeout)
-			ctx = c
-			defer cancel()
```

Comment:
one thing that isn't clear to me - why do we cancel outside of scrape now instead of inside?

If we do need to, since its not obvious can we explain why (in code comment)?

Reply:
does `cancel()` mean the `res.Body` will be also be closed (before we read it)?

Reply:
The unit tests seem to pass if we keep the `defer cancel()`, so maybe its possible or we don't have enough coverage.

Reply:
Reading `res.Body()` after `cancel()`,  it flakes. Maybe success when body's size is small.
See discussion on https://groups.google.com/g/golang-nuts/c/2FKwG6oEvos

Reply:
We should add a unit test to cover big response body.